src/client-agent/buffer.c

```diff
@@ -266,3 +274,111 @@ int w_agentd_get_buffer_lenght() {
 
     return retval;
 }
+
+void w_agentd_buffer_free(unsigned int current_capacity) {
+    w_mutex_lock(&mutex_lock);
+
+    // Ensure the buffer is actually allocated before trying to free.
+    if ( buffer == NULL || current_capacity == 0) {
+        mwarn("Buffer is already unallocated or invalid. Skipping free operation.");
+        w_mutex_unlock(&mutex_lock);
+        return;
+    }
+
+    mdebug2("Freeing the client-buffer.");
+    for ( int i=0; i <= current_capacity; i++) {
+        if (buffer[i] != NULL ) os_free(buffer[i]);
```

Comment:
```suggestion
        os_free(buffer[i]);
```

`os_free()` internally already checks that `buffer[i]` contains a value, so we can skip the check.